src/plugins/visualize/public/application/types.ts

```diff
@@ -47,29 +57,59 @@ export interface VisualizeAppStateTransitions {
     value: VisualizeAppState[T]
   ) => VisualizeAppState;
   setVis: (state: VisualizeAppState) => (vis: Partial<PureVisState>) => VisualizeAppState;
-  removeSavedQuery: (state: VisualizeAppState) => (defaultQuery: Query) => VisualizeAppState;
   unlinkSavedSearch: (
     state: VisualizeAppState
   ) => ({ query, parentFilters }: { query?: Query; parentFilters?: Filter[] }) => VisualizeAppState;
   updateVisState: (state: VisualizeAppState) => (vis: PureVisState) => VisualizeAppState;
-  updateFromSavedQuery: (state: VisualizeAppState) => (savedQuery: SavedQuery) => VisualizeAppState;
+  updateSavedQuery: (state: VisualizeAppState) => (savedQueryId?: string) => VisualizeAppState;
 }
 
+export type VisualizeAppStateContainer = ReduxLikeStateContainer<
+  VisualizeAppState,
+  VisualizeAppStateTransitions
+>;
+
 export interface EditorRenderProps {
   core: CoreStart;
   data: DataPublicPluginStart;
   filters: Filter[];
   timeRange: TimeRange;
   query?: Query;
-  savedSearch?: SavedSearch;
+  savedSearch?: SavedObject | null;
```

Comment:
Are you sure we need to allow passing `null` value? I found only one place where you set null value `src/plugins/visualize/public/application/utils/get_visualization_instance.ts`. Looks like it can be easily fixed

Reply:
You're right! Thanks for the notice! Done.